File: CodeForces/Java/CF_344A_Magnets.java
//  Problem: 344A. Magnets
//  Link: https://codeforces.com/problemset/problem/344/A

import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    int n = scan.nextInt();
    int groups = 1;
  
    int prev = scan.nextInt();
    
    for (int i = 1; i < n; i++) {
        int current = scan.nextInt();
        if (current != prev) {
            groups++;
            prev = current;
        }
    }
    System.out.println(groups);
    
    scan.close();
  }
}

File: CodeForces/Python/CF_344A_Magnets.py
# Problem: 344A. Magnets
# Link: https://codeforces.com/problemset/problem/344/A

n = int(input())

groups = 1
prev = input()

for _ in range(n - 1):
    current = input()
    if current != prev:
        groups += 1
        prev = current

print(groups)
